echopype/calibrate/ek80_complex.py

```diff
@@ -323,18 +279,21 @@ def compress_pulse(backscatter: xr.DataArray, chirp: Dict) -> xr.DataArray:
         # Select channel `chan` and drop the specific beam dimension if all of the values are nan.
         backscatter_chan = backscatter.sel(channel=chan).dropna(dim="beam", how="all")
 
+        # Create NaN mask
+        nan_mask = np.isnan(backscatter_chan.compute())
```

Comment:
Do we need `.compute` here?

Reply:
Just tested it and unfortunately no. Without the `.compute()`, the `xr.where` at line 286, and the one at line 306 doesn't force compute it. Without computing prior, line 306 will turn `pc` from something computed to something not computed.

Reply:
If you only have .compute() in the line with xr.where, does it not work?

Reply:
I guess it does not really matter in this situation, where we have .compute somewhere in the middle of the function... maybe when we need to figure out how to distribute efficiently....

Reply:
It does work, made that change👍

Reply:
What I was thinking was to do 
```
pc=xr.where(nan_mask, np.nan,pc)
pc_all.append(pc.compute())
```
But looking at the code again, I think it'll actually be fine to keep things not computed until users explicitly want it to be computed. In a sense it is how it's supposed to be, like how we have it in `combine_echodata`. We can address that in another go.